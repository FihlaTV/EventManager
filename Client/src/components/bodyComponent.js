import React from 'react';
import { RecoverPassword } from './rePasswordComponent';
import { SignIn } from './signinComponent'

export const Content = (props) => {
    return (
      <div>
        <SignIn />
        <RecoverPassword />
      </div>
    );
  }
  