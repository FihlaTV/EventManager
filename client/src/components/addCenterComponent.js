import React from 'react';
import { ModalHeader } from './modalHeader';
import { FormGroup } from './formGroup';
import { Option } from './selectOption';
import { apiLink, Transactions } from '../services';
import { connect } from 'react-redux';
import { setCenterDetails } from '../actions/centerActions';
import { setSubmitState } from '../actions/submitAction';
import { setCenterDefaults } from '../actions/pageActions';
import PropTypes from 'prop-types';
import { ModalFooter } from './modalFooter';
import { initialState } from '../reducers/pageReducer';

const inputAttrs = (inputType, inputName, placeholder, className, ref, required, autocomplete) => ({
  inputType,
  inputName,
  placeholder,
  className,
  ref,
  required,
  autocomplete,
});

const mapDispatchToProps = dispatch => ({
  setCenterDetails: center => dispatch(setCenterDetails(center)),
  setSubmitState: submitState => dispatch(setSubmitState(submitState)),
  setCenterDefaults: submitState => dispatch(setCenterDefaults(submitState)),
});

const mapStateToProps = state => ({
  modalTitle: state.page.modalTitle,
  required: state.page.required,
  centerDefaults: state.page.centerDefaults,
  disabled: state.process.disabled,
  visibility: state.process.visibility,
});

let centerId;
const folder = apiLink === 'http://localhost:8080' ? 'dev/centers' : 'prod/centers';
export class AddCenterComponent extends React.Component {
  handleSubmit = event => {
    event.preventDefault();
    const props = this.props;
    props.setSubmitState('processing');
    const closeModal = this.closeModal;
    const transactions = new Transactions(this.props, 'center');
    const saveCenter = res => {
      const centerData = {
        name: this.name.value,
        address: this.address.value,
        location: this.location.value,
        capacity: this.capacity.value,
        price: this.price.value,
        availability: this.availability.value,
        token: JSON.parse(localStorage.token).value,
        picture: res ? res.data.secure_url : undefined,
        publicId: res ? res.data.public_id : undefined,
      };
      transactions.addOrUpdate(centerId, centerData, err => {
        props.setSubmitState('initial');
        if (!err) {
          closeModal();
        }
      });
    };
    if (this.picture.files[0]) {
      const imageData = new FormData();
      const publicId = `${Date.now()}-${this.picture.files[0].name}`;
      imageData.append('file', this.picture.files[0]);
      imageData.append('tags', 'center, facilities, events');
      imageData.append('upload_preset', `${process.env.UPLOAD_PRESET}`);
      imageData.append('api_key', `${process.env.API_KEY}`);
      imageData.append('timestamp', (Date.now() / 1000) | 0);
      imageData.append('folder', folder);
      imageData.append('public_id', publicId);
      transactions.uploadImage(imageData, saveCenter, () => {
        props.setSubmitState('initial');
      });
    } else {
      saveCenter(undefined);
    }
  };
  closeModal = () => {
    this.picture.value = '';
    this.props.setCenterDefaults(initialState);
  };
  componentWillReceiveProps(nextState) {
    const centerDefaults = nextState.centerDefaults;
    if (centerDefaults.name !== null) {
      this.name.value = centerDefaults.name || '';
      this.address.value = centerDefaults.address || '';
      this.location.value = centerDefaults.location || '';
      this.capacity.value = centerDefaults.capacity;
      this.price.value = centerDefaults.price;
      this.availability.value = centerDefaults.availability;
    }
    centerId = centerDefaults.id;
  }
  render() {
    const { modalTitle, required } = this.props;
    return (
      <div
        className="modal fade"
        role="dialog"
        id="addNewCenter"
        tabIndex="-1"
        aria-labelledby="addNewCenterLabel"
        aria-hidden="true"
        data-keyboard="false"
        data-backdrop="static"
      >
        <div className="modal-dialog">
          <div className="modal-content eventModal">
            <ModalHeader id="addNewCenterTitle" title={modalTitle} action={this.closeModal} />
            <div className="modal-body mx-sm-auto col-sm-10">
              <form role="form" onSubmit={this.handleSubmit}>
                <FormGroup
                  image="glyphicons-21-home.png"
                  alt="centername"
                  inputProps={inputAttrs(
                    'text',
                    'centername',
                    'Center Name',
                    'form-control input-sm',
                    input => (this.name = input),
                    'required'
                  )}
                />
                <FormGroup
                  image="glyphicons-139-picture.png"
                  alt="centerImage"
                  inputProps={inputAttrs(
                    'file',
                    'centerImage',
                    'Center Image',
                    'form-control input-sm',
                    input => (this.picture = input),
                    required
                  )}
                />
                <FormGroup
                  image="glyphicons-243-map-marker.png"
                  alt="address"
                  inputProps={inputAttrs(
                    'text',
                    'street',
                    'Address',
                    'form-control input-sm',
                    input => (this.address = input),
                    'required',
                    'street-address'
                  )}
                />
                <FormGroup
                  image="glyphicons-740-cadastral-map.png"
                  alt="city"
                  inputProps={inputAttrs(
                    'text',
                    'city',
                    'State/City',
                    'form-control input-sm',
                    input => (this.location = input),
                    'required',
                    'address-level2'
                  )}
                />
                <FormGroup
                  image="glyphicons-44-group.png"
                  alt="capacity"
                  inputProps={inputAttrs(
                    'number',
                    'capacity',
                    'Capacity',
                    'form-control input-sm',
                    input => (this.capacity = input),
                    'required'
                  )}
                />
                <FormGroup
                  image="glyphicons-228-usd.png"
                  alt="bookingAmount"
                  inputProps={inputAttrs(
                    'number',
                    'bookingAmount',
                    'Booking price(per day)',
                    'form-control input-sm',
                    input => (this.price = input),
                    'required'
                  )}
                />
                <div className="form-group">
                  <select
                    required
                    ref={input => (this.availability = input)}
                    className="custom-select-sm"
                  >
                    <Option value="" text="Availability" disabled selected />
                    <Option value="open" text="open" />
                    <Option value="close" text="close" />
                  </select>
                </div>
                <ModalFooter
                  type="submit"
                  disabled={this.props.disabled}
                  display={this.props.visibility}
                  closeModal={this.closeModal}
                />
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export const AddCenter = connect(mapStateToProps, mapDispatchToProps)(AddCenterComponent);
AddCenterComponent.propTypes = {
  modalTitle: PropTypes.string,
  required: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  visibility: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
};
