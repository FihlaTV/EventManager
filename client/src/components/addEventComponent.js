import React from 'react';
import { FormGroup } from './formGroup';
import { ModalHeader } from './modalHeader';
import { Option } from './selectOption';
import { apiLink, Transactions, getAll, toastSettings } from '../services';
import { connect } from 'react-redux';
import { setEventDetail } from '../actions/eventActions';
import { getEventCenters } from '../actions/centerActions';
import { setDataCount, setActivePage, setEventDefaults } from '../actions/pageActions';
import { setSubmitState } from '../actions/submitAction';
import PropTypes from 'prop-types';
import { TableRow } from './table';
import Pagination from 'react-js-pagination';
import toastr from 'toastr';
import { ModalFooter } from './modalFooter';
import { initialState } from '../reducers/pageReducer';

toastr.options = toastSettings;
const inputAttrs = (inputType, inputName, placeholder, className, ref, id, required) => ({
  inputType,
  inputName,
  placeholder,
  className,
  ref,
  id,
  required,
});

let eventId;
const folder = apiLink === 'http://localhost:8080' ? 'dev/events' : 'prod/events';
export class AddEventComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedClass: 'custom-select-sm btn btn-info',
      displayDate: 'none',
      displayTime: 'none',
    };
  }
  checkValidation = () => {
    if (this.center.value === '') {
      toastr.error('No center selected');
    }
  };

  closeModal = () => {
    this.setState({
      selectedClass: 'custom-select-sm btn btn-info',
      displayDate: 'none',
      displayTime: 'none',
    });
    this.props.setEventDefaults(initialState);
    this.picture.value = '';
  };

  selectCenter = e => {
    this.setState({
      selectedClass: 'custom-select-sm btn btn-success',
    });
    this.center.value = e.target.id;
  };

  handlePageChange = pageNumber => {
    this.setState({
      selectedClass: 'custom-select-sm btn btn-info',
    });
    getAll(this.props, 'eventCenters', pageNumber, 1);
    this.props.setActivePage(pageNumber);
  };

  handleSubmit = event => {
    event.preventDefault();
    const props = this.props;
    const closeModal = this.closeModal;
    props.setSubmitState('processing');
    const transactions = new Transactions(this.props, 'event');
    const title = this.name.value;
    const date = this.date.value;
    const time = this.time.value;
    const description = this.description.value;
    const centerId = this.center.value;
    const token = JSON.parse(localStorage.token).value;
    function saveEvent(res) {
      const eventData = {
        title,
        picture: res ? res.data.secure_url : undefined,
        publicId: res ? res.data.public_id : undefined,
        date,
        time,
        description,
        centerId,
        token,
      };
      transactions.addOrUpdate(eventId, eventData, err => {
        props.setSubmitState('initial');
        if (!err) {
          closeModal();
        }
      });
    }
    if (this.picture.files[0]) {
      const imageData = new FormData();
      const publicId = `${Date.now()}-${this.picture.files[0].name}`;
      imageData.append('file', this.picture.files[0]);
      imageData.append('tags', 'center, facilities, events');
      imageData.append('upload_preset', `${process.env.UPLOAD_PRESET}`);
      imageData.append('api_key', `${process.env.API_KEY}`);
      imageData.append('timestamp', (Date.now() / 1000) | 0);
      imageData.append('folder', folder);
      imageData.append('public_id', publicId);
      transactions.uploadImage(imageData, saveEvent, () => {
        props.setSubmitState('initial');
      });
    } else {
      saveEvent(undefined);
    }
  };

  generateDate = eventDefaults => {
    const date = new Date(eventDefaults.date); // The Date object lets you work with dates.
    const year = date.getFullYear(); // This method gets the four digit year.
    let month = date.getMonth() + 1; // This method gets the month and Jan is 0.
    let day = date.getDate(); // This method gets the day of month as a number.
    let hour = date.getHours(); // This method gets the hour
    let min = date.getMinutes(); // This method gets the minutes
    month = month < 10 ? `0${month}` : month;
    day = day < 10 ? `0${day}` : day;
    hour = hour < 10 ? `0${hour}` : hour;
    // It adds a 0 to number less than 10 because input[type=time] only accepts 00:00 format.
    min = min < 10 ? `0${min}` : min;
    const eventDate = `${year}-${month}-${day}`;
    const eventTime = `${hour}:${min}`;
    return { eventDate, eventTime };
  };

  // Bind input controls with data when user wants to modify event
  // use get_derived_state_from_props
  componentWillReceiveProps(nextState) {
    const eventDefaults = nextState.eventDefaults;
    if (eventDefaults.title !== null) {
      eventId = eventId || eventDefaults.id;
      this.name.value = eventDefaults.title || '';
      const dateObject = this.generateDate(eventDefaults);
      this.date.value = dateObject.eventDate;
      this.time.value = dateObject.eventTime;
      this.description.value = eventDefaults.description || '';
      this.center = eventDefaults.centerId;
    }
  }

  handleChange = () => {
    const data = {
      title: this.name.value,
      date: new Date(`${this.date.value} ${this.time.value}`),
      picture: this.picture.value,
      description: this.description.value,
      centerId: this.center.value,
    };
    if (this.date.value !== '') {
      this.setState({
        displayDate: 'block',
      });
    }
    if (this.time.value !== '') {
      this.setState({
        displayTime: 'block',
      });
    }
    this.props.setEventDefaults(data);
  };

  render() {
    const { modalTitle, centers } = this.props;
    return (
      <div
        className="modal fade"
        role="dialog"
        id="addNewEvent"
        tabIndex="-1"
        aria-labelledby="addNewEventLabel"
        aria-hidden="true"
        data-keyboard="false"
        data-backdrop="static"
      >
        <div className="modal-dialog">
          <div className="modal-content eventModal">
            <ModalHeader id="addNewEventTitle" title={modalTitle} action={this.closeModal} />
            <div className="modal-body mx-sm-auto col-sm-10">
              <form role="form" id="eventForm" onSubmit={this.handleSubmit}>
                <FormGroup
                  image="glyphicons-619-mixed-buildings.png"
                  alt="eventname"
                  onChange={this.handleChange}
                  inputProps={inputAttrs(
                    'text',
                    'eventTitle',
                    'Event Title',
                    'form-control input-sm',
                    input => (this.name = input),
                    'required'
                  )}
                />
                <FormGroup
                  image="glyphicons-139-picture.png"
                  alt="eventImage"
                  onChange={this.handleChange}
                  inputProps={inputAttrs(
                    'file',
                    'eventImage',
                    'Event Image',
                    'form-control input-sm',
                    input => (this.picture = input)
                  )}
                />
                <FormGroup
                  image="glyphicons-46-calendar.png"
                  alt="eventdate"
                  onChange={this.handleChange}
                  inputProps={inputAttrs(
                    'date',
                    'eventdate',
                    'Event Date',
                    'form-control input-sm',
                    input => (this.date = input),
                    'event-date',
                    'required'
                  )}
                />
                <FormGroup
                  image="glyphicons-54-alarm.png"
                  alt="eventTime"
                  onChange={this.handleChange}
                  inputProps={inputAttrs(
                    'time',
                    'eventTime',
                    'Event Time',
                    'form-control input-sm',
                    input => (this.time = input),
                    'event-time',
                    'required'
                  )}
                />
                <div className="form-group">
                  <label id="description" htmlFor="description" className="control-label">
                    Description
                  </label>
                  <textarea
                    required
                    name="description"
                    ref={input => (this.description = input)}
                    rows="2"
                    className="form-control"
                    onChange={this.handleChange}
                  />
                </div>
                <div className="form-group">
                  <select
                    required
                    ref={input => (this.center = input)}
                    className="custom-select-sm"
                    style={{ display: 'none' }}
                    onChange={this.handleChange}
                  >
                    <Option value="" text="Select Center" disabled selected />
                    {centers.map(center => (
                      <Option
                        key={center.id}
                        value={center.id}
                        text={`${center.name}, ${center.address}, ${center.location}`}
                      />
                    ))}
                  </select>
                  {centers.map(center => (
                    <div key={center.id}>
                      <div>
                        <div className="event-center-img">
                          <img
                            id="cardImage"
                            className="event-center-Image"
                            src={`${center.picture}`}
                            alt="centerImage"
                          />
                        </div>
                        <div>
                          <div className="table-responsive">
                            <table
                              style={{ border: 'none' }}
                              className="table table-hover table-striped table-bordered"
                            >
                              <tbody>
                                <TableRow
                                  columns={[
                                    'name',
                                    center.availability === 'open' ? (
                                      <p>{center.name}</p>
                                    ) : (
                                      <p className="rejected">{center.name} (closed)</p>
                                    ),
                                  ]}
                                />
                                <TableRow
                                  columns={['Location', `${center.address} ${center.location}`]}
                                />
                                <TableRow columns={['Capacity', center.capacity]} />
                              </tbody>
                            </table>
                          </div>
                        </div>
                      </div>
                      <button
                        id={center.id}
                        type="button"
                        onClick={this.selectCenter}
                        className={this.state.selectedClass}
                        style={{ borderRadius: '0em' }}
                      >
                        select
                      </button>
                      &nbsp;
                    </div>
                  ))}
                  <Pagination
                    activePage={this.props.activePage}
                    innerClass="pagination add-event"
                    itemsCountPerPage={1}
                    totalItemsCount={this.props.dataCount}
                    pageRangeDisplayed={5}
                    onChange={this.handlePageChange}
                    itemClass="page-item"
                    linkClass="page-link"
                  />
                  <ModalFooter
                    type="submit"
                    disabled={this.props.disabled}
                    checkValidation={this.checkValidation}
                    display={this.props.visibility}
                    closeModal={this.closeModal}
                  />
                  <p id="displayDate" style={{ display: this.state.displayDate }}>
                    {' '}
                  </p>
                  <p id="displayTime" style={{ display: this.state.displayTime }}>
                    {' '}
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const actionCreators = {
  setEventDetail,
  getEventCenters,
  setDataCount,
  setSubmitState,
  setActivePage,
  setEventDefaults,
};
const mapStateToProps = state => ({
  modalTitle: state.page.modalTitle,
  eventDetails: state.events.event,
  eventDefaults: state.page.eventDefaults,
  centers: state.centers.eventCenters,
  dataCount: state.page.dataCount,
  disabled: state.process.disabled,
  visibility: state.process.visibility,
  activePage: state.page.activePage,
  random: state.page.random,
});

export const AddEvent = connect(mapStateToProps, actionCreators)(AddEventComponent);
AddEventComponent.propTypes = {
  modalTitle: PropTypes.string,
  centers: PropTypes.arrayOf(PropTypes.object),
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  visibility: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  dataCount: PropTypes.number,
  match: PropTypes.object,
  activePage: PropTypes.number,
  setActivePage: PropTypes.func,
};
