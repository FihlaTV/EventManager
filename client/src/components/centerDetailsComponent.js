import React from 'react';
import { ManageDetailsHeader } from './manageDetailsHeader';
import { TableHead, TableRow } from './table';
import { setCenterDetails } from '../actions/centerActions';
import { setPage } from '../actions/pageActions';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getOne } from '../services';
import Icon from './icon';
import { ManageFacilities } from './manageFacilityComponent';

const mapDispatchToProps = dispatch => ({
  setCenterDetails: center => dispatch(setCenterDetails(center)),
  setPage: page => dispatch(setPage(page)),
});

const mapStateToProps = state => ({
  centerDetails: state.centers.centerDetails,
  facilities: state.facilities.facilities,
});

class CenterDetailsComponent extends React.Component {
  constructor(props) {
    super(props);
    this.id = this.props.match.params.id;
  }
  componentWillMount() {
    getOne(this.props, this.id, 'centers');
  }
  componentWillUnmount() {
    this.props.setPage('dashboard');
  }
  render() {
    const center = this.props.centerDetails[0];
    if (center === undefined) {
      return (
        <div>
          <h2>Loading...</h2>
        </div>
      );
    }
    const loggedInUser = JSON.parse(localStorage.token).id;
    const centerOwner = center.userId;
    const facilities = this.props.facilities.length > 0 ? this.props.facilities : center.facilities;
    return (
      <div className="card mx-sm-auto col-sm-11 zero-padding">
        {loggedInUser === centerOwner && (
          <ManageFacilities history={this.props.history} data={facilities} centerId={this.id} />
        )}
        <div className="card-header mg-event-header card-header-body">
          <ManageDetailsHeader
            history={this.props.history}
            param={this.id}
            title={center.name}
            editModal="#addNewCenter"
          />
        </div>
        <div className="card-body">
          <div className="row">
            <div className="col-sm-5">
              <img
                id="cardImage"
                className="card-image"
                src={`${center.picture}`}
                alt="centerImage"
              />
            </div>
            <div className="col-sm-7">
              <div className="table-responsive">
                <table
                  style={{ border: 'none' }}
                  className="table table-hover table-striped table-bordered"
                >
                  <tbody>
                    <TableRow columns={['Street Address', center.address]} />
                    <TableRow columns={['City', center.location]} />
                    <TableRow columns={['Capacity', center.capacity]} />
                    <TableRow columns={['Booking price', center.price]} />
                    <TableRow columns={['Availability', center.availability]} />
                    <TableRow columns={['Contact', center.user.email]} />
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <br />
          <div className="top-eight add-facility">
            <div className="col-sm-10 zero-padding add-facility-90">
              <b className="appTitle">Facilities</b>
            </div>
            {loggedInUser === centerOwner && (
              <div className="col-sm-2 zero-padding" id="addFacility">
                <button
                  type="submit"
                  id="manageFacty"
                  className="btn btn-success manageFacility"
                  data-toggle="modal"
                  data-target="#manageFacilities"
                >
                  <Icon src="glyphicons-281-settings.png" alt="Edit" />
                </button>
              </div>
            )}
          </div>
          <div className="table-responsive">
            <table className="table table-hover table-fixed table-striped">
              <TableHead columns={['Name', 'Spec', 'Quantity']} class="table-header" />
              <tbody>
                {facilities.map(facility => (
                  <TableRow
                    key={facility.id}
                    columns={[
                      <b>{facility.name}</b>,
                      <b>{facility.spec}</b>,
                      <span className="badge">{facility.quantity}</span>,
                    ]}
                  />
                ))}
              </tbody>
            </table>
          </div>
          <br />
          <div className="table-responsive">
            <table className="table table-hover table-fixed table-striped">
              <TableHead columns={['Events', 'Title', 'Date']} class="table-header" />
              <tbody>
                {center.events.map(event => (
                  <TableRow
                    key={event.id}
                    columns={[
                      <img className="center-image" src={`${event.picture}`} alt="event-view" />,
                      <b>
                        <Link className="event-detail" to={`/dashboard/events/${event.id}`}>
                          {event.title}
                        </Link>
                      </b>,
                      new Date(event.date).toDateString(),
                    ]}
                  />
                ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    );
  }
}
export const CenterDetails = connect(mapStateToProps, mapDispatchToProps)(CenterDetailsComponent);
