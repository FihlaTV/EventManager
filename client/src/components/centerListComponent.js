import React from 'react';
import { TableHead, TableRow } from './table';
import { Link } from 'react-router-dom';
import { populateCenters } from '../actions/centerActions';
import { setDataCount, setActivePage } from '../actions/pageActions';
import { connect } from 'react-redux';
import { getAll, searchFunction } from '../services';
import PropTypes from 'prop-types';
import Pagination from 'react-js-pagination';
import { LIMIT } from '../constants/actionTypes';
import { Filter } from './filterComponent';

const mapDispatchToProps = dispatch => ({
  populateCenters: centers => dispatch(populateCenters(centers)),
  setDataCount: count => dispatch(setDataCount(count)),
  setActivePage: pageNumber => dispatch(setActivePage(pageNumber)),
});
const mapStateToProps = state => ({
  centers: state.centers.centerList,
  dataCount: state.page.dataCount,
  activePage: state.page.activePage,
});

export class CenterList extends React.Component {
  searchEvents = e => {
    searchFunction(e, 'centerTable');
  };

  componentWillMount() {
    getAll(this.props, 'centers');
  }

  handlePageChange = pageNumber => {
    this.props.setActivePage(pageNumber);
    getAll(this.props, 'centers', pageNumber);
  };

  render() {
    return (
      <div className="mx-sm-auto col-sm-11">
        <b className="page-header">Centers</b>
        <Filter
          id="filter-centers"
          placeholder="Filter by name or location..."
          handleSearch={this.searchEvents}
        />
        <div className="table-responsive">
          <table id="centerTable" className="table table-hover table-main">
            <TableHead
              columns={['View', 'Name', 'Location', 'Capacity']}
              class="table-header table-header-main"
            />
            <tbody>
              {this.props.centers.map(center => (
                <TableRow
                  key={center.id}
                  columns={[
                    <img
                      key="nbhsidj0"
                      className="center-image"
                      src={`${center.picture}`}
                      alt="center-view"
                    />,
                    center.availability === 'open' ? (
                      <Link className="event-detail" to={`${this.props.match.path}/${center.id}`}>
                        {center.name}
                      </Link>
                    ) : (
                      <Link
                        className="event-detail rejected"
                        to={`${this.props.match.path}/${center.id}`}
                      >
                        {center.name} (closed)
                      </Link>
                    ),
                    `${center.name}, ${center.address}`,
                    <span key="centerCapacity1782yhbj" className="badge">
                      {center.capacity}
                    </span>,
                  ]}
                />
              ))}
            </tbody>
          </table>
          <br />
          <Pagination
            activePage={this.props.activePage}
            itemsCountPerPage={LIMIT}
            totalItemsCount={this.props.dataCount}
            pageRangeDisplayed={3}
            onChange={this.handlePageChange}
            itemClass="page-item"
            linkClass="page-link"
          />
        </div>
      </div>
    );
  }
}

export const MyCenters = connect(mapStateToProps, mapDispatchToProps)(CenterList);
CenterList.propTypes = {
  centers: PropTypes.arrayOf(PropTypes.object),
  match: PropTypes.object,
  activePage: PropTypes.number,
  dataCount: PropTypes.number,
  setActivePage: PropTypes.func,
};
