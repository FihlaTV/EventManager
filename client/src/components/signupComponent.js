import React from 'react';
import { FormGroup } from './formGroup';
import { connect } from 'react-redux';
import { setAccountType } from '../actions/userActions';
import { onboarding, toastSettings, userValidator } from '../services';
import toastr from 'toastr';
import PropTypes from 'prop-types';

toastr.options = toastSettings;

const inputAttrs = (inputType, inputName, placeholder, className, ref, required) => ({
  inputType,
  inputName,
  placeholder,
  className,
  ref,
  required,
});
// all props are from signin component
export class SignupComponent extends React.Component {
  handleSubmit = event => {
    event.preventDefault();
    const signupData = {
      username: this.username.value,
      password: this.password.value,
      name: this.name.value,
      confirmPassword: this.rePassword.value,
      email: this.email.value,
      phoneNo: `+${this.phone.value}`,
      accountType: 'regular',
    };
    const validationStatus = userValidator(signupData, 'signup');
    if (validationStatus === true) {
      this.props.changeSubmitState('processing');
      onboarding(this.props, signupData, 'signup', () => {
        this.props.changeSubmitState('initial');
      });
    } else {
      toastr.error(validationStatus);
    }
  };

  componentWillUnmount() {
    this.props.changeSubmitState('initial');
  }

  render() {
    return (
      <form role="form" onSubmit={this.handleSubmit} className="formDiv" id="signupForm">
        <h3 className="text-center panel-font">
          <b>Register</b>
        </h3>
        <br />
        <FormGroup
          image="glyphicons-522-user-lock.png"
          alt="username"
          inputProps={inputAttrs(
            'text',
            'username',
            'Username',
            'form-control input-sm',
            input => (this.username = input),
            'required'
          )}
        />
        <FormGroup
          image="glyphicons-4-user.png"
          alt="fullname"
          inputProps={inputAttrs(
            'text',
            'fullname',
            'Full name',
            'form-control input-sm',
            input => (this.name = input),
            'required'
          )}
        />
        <FormGroup
          image="glyphicons-11-envelope.png"
          alt="email"
          inputProps={inputAttrs(
            'email',
            'email',
            'Email',
            'form-control input-sm',
            input => (this.email = input),
            'required'
          )}
        />
        <FormGroup
          image="glyphicons-442-phone-alt.png"
          alt="phone"
          inputProps={inputAttrs(
            'number',
            'phone',
            'Phone number',
            'form-control input-sm',
            input => (this.phone = input),
            'required'
          )}
        />
        <FormGroup
          image="glyphicons-204-lock.png"
          alt="password"
          inputProps={inputAttrs(
            'password',
            'password',
            'Password',
            'form-control input-sm',
            input => (this.password = input),
            'required'
          )}
        />
        <FormGroup
          image="glyphicons-204-lock.png"
          alt="password"
          inputProps={inputAttrs(
            'password',
            'rePassword',
            'Re-enter Password',
            'form-control input-sm',
            input => (this.rePassword = input),
            'required'
          )}
        />
        <button
          type="submit"
          disabled={this.props.disabled}
          className="btn btn-lg btn-primary btn-block submitButton"
        >
          <i className="fa fa-spinner fa-spin" style={{ display: this.props.visibility }} />
          &nbsp; Signup
        </button>
        <div className="form-links">
          <a href="#" className="welcome" onClick={this.props.changeState}>
            Already have an account? Signin
          </a>
        </div>
      </form>
    );
  }
}
export const mapDispatchToProps = dispatch => ({
  setAccountType: accountType => dispatch(setAccountType(accountType)),
});
const validations = {
  changeSubmitState: PropTypes.func,
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  visibility: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  changeState: PropTypes.func,
};
SignupComponent.propTypes = validations;

export const SignupForm = connect(null, mapDispatchToProps)(SignupComponent);
