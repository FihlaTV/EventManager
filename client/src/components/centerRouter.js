import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { MyCenters } from './centerListComponent';
import { CenterDetails } from './centerDetailsComponent';

export class CenterRouter extends React.Component {
  render() {
    return (
      <Switch>
        <Route path={`${this.props.match.path}/:id`} component={CenterDetails} />
        <Route path={`${this.props.match.path}`} component={MyCenters} />
      </Switch>
    );
  }
}
