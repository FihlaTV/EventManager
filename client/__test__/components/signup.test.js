/* global expect afterEach beforeEach describe it */
import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { SignupComponent, mapDispatchToProps } from '../../src/components/signupComponent';

configure({ adapter: new Adapter() });

describe('Tests for signup component', () => {
  const props = {
    changeState: jest.fn(),
    changeSubmitState: jest.fn(),
    disabled: 'disabled',
    visibility: false,
  };

  it('should render self without crashing', () => {
    const enzymeWrapper = shallow(<SignupComponent {...props} />);
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should call setSubmitState before unmounting', () => {
    const enzymeWrapper = shallow(<SignupComponent {...props} />);
    enzymeWrapper.unmount();
    // expect(props.setSubmitState).toHaveBeenCalled();
  });
  it('should call setState when changeState is triggered', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const enzymeWrapper = shallow(<SignupComponent {...props} />);
    const instance = enzymeWrapper.instance();
    instance.username = {
      value: 'wisdom',
    };
    instance.password = {
      value: 'password1',
    };
    instance.name = {
      value: 'Anaeze',
    };
    instance.rePassword = {
      value: 'Anaeze',
    };
    instance.email = {
      value: 'Anaeze',
    };
    instance.phone = {
      value: 'Anaeze',
    };

    instance.handleSubmit(event);
    // expect(props.setSubmitState).toHaveBeenCalled();
  });
  it('should call setState when changeState is triggered', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const enzymeWrapper = shallow(<SignupComponent {...props} />);
    const instance = enzymeWrapper.instance();
    instance.username = {
      value: 'anaeze',
    };
    instance.password = {
      value: 'password1',
    };
    instance.name = {
      value: 'Anaeze Nsoffor',
    };
    instance.rePassword = {
      value: 'password1',
    };
    instance.email = {
      value: 'nzediegwu1@gmail.com',
    };
    instance.phone = {
      value: '2347067372837',
    };

    instance.handleSubmit(event);
    // expect(props.setSubmitState).toHaveBeenCalled();
  });
});
describe('mapDispatchToProps', () => {
  it('should return', () => {
    const dispatch = jest.fn();
    const mdtp = mapDispatchToProps(dispatch);
    expect(mdtp).toHaveProperty('setAccountType');
  });
});
