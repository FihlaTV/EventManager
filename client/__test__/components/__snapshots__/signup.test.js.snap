// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for signup component should render self without crashing 1`] = `
<form
  className="formDiv"
  id="signupForm"
  onSubmit={[Function]}
  role="form"
>
  <h3
    className="text-center panel-font"
  >
    <b>
      Register
    </b>
  </h3>
  <br />
  <FormGroup
    alt="username"
    image="glyphicons-522-user-lock.png"
    inputProps={
      Object {
        "className": "form-control input-sm",
        "inputName": "username",
        "inputType": "text",
        "placeholder": "Username",
        "ref": [Function],
        "required": "required",
      }
    }
  />
  <FormGroup
    alt="fullname"
    image="glyphicons-4-user.png"
    inputProps={
      Object {
        "className": "form-control input-sm",
        "inputName": "fullname",
        "inputType": "text",
        "placeholder": "Full name",
        "ref": [Function],
        "required": "required",
      }
    }
  />
  <FormGroup
    alt="email"
    image="glyphicons-11-envelope.png"
    inputProps={
      Object {
        "className": "form-control input-sm",
        "inputName": "email",
        "inputType": "email",
        "placeholder": "Email",
        "ref": [Function],
        "required": "required",
      }
    }
  />
  <FormGroup
    alt="phone"
    image="glyphicons-442-phone-alt.png"
    inputProps={
      Object {
        "className": "form-control input-sm",
        "inputName": "phone",
        "inputType": "number",
        "placeholder": "Phone number",
        "ref": [Function],
        "required": "required",
      }
    }
  />
  <FormGroup
    alt="password"
    image="glyphicons-204-lock.png"
    inputProps={
      Object {
        "className": "form-control input-sm",
        "inputName": "password",
        "inputType": "password",
        "placeholder": "Password",
        "ref": [Function],
        "required": "required",
      }
    }
  />
  <FormGroup
    alt="password"
    image="glyphicons-204-lock.png"
    inputProps={
      Object {
        "className": "form-control input-sm",
        "inputName": "rePassword",
        "inputType": "password",
        "placeholder": "Re-enter Password",
        "ref": [Function],
        "required": "required",
      }
    }
  />
  <button
    className="btn btn-lg btn-primary btn-block submitButton"
    disabled="disabled"
    type="submit"
  >
    <i
      className="fa fa-spinner fa-spin"
      style={
        Object {
          "display": false,
        }
      }
    />
    Â  Signup
  </button>
  <div
    className="form-links"
  >
    <a
      className="welcome"
      href="#"
      onClick={[MockFunction]}
    >
      Already have an account? Signin
    </a>
  </div>
</form>
`;
